wrapper/src/main/java/software/amazon/jdbc/DriverConnectionProvider.java

```diff
@@ -114,9 +114,11 @@ public Connection connect(
       final @NonNull Properties props)
       throws SQLException {
 
-    final Properties copy = PropertyUtils.copyProperties(props);
+    final Properties copiedProps = PropertyUtils.copyProperties(props);
+    PropertyUtils.addConnAttrForClientIdentification(dialect, copiedProps);
```

Comment:
Need to call 
dialect.prepareProperties(copiedProps)

See my comments for PropertyUtils

Reply:
Need to add similar changes to DataSourceConnectionProvider